File: 2025_01/leetcode2025-01-15.py
from typing import List
from heapq import *

class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        pq = nums
        heapify(pq)
        res = 0
        while len(pq)>=2 and pq[0]<k:
            x = heappop(pq)
            y = heappop(pq)
            heappush(pq, x*2+y)
            res += 1
        return -1 if pq[0]<k else res

if __name__ == "__main__":
    s = Solution()
    nums = [2,11,10,1,3]
    k = 10
    print(s.minOperations(nums, k))

File: leetcode2025/2025_01/leetcode2025-01-15.cpp
#include <iostream>
#include <vector>
#include <queue>
using namespace std;

class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        priority_queue<long long, vector<long long>, greater<long long>> pq(nums.begin(), nums.end());
        int res = 0;
        while (pq.size()>=2 && pq.top()<k){
            long long x = pq.top();
            pq.pop();
            long long y = pq.top();
            pq.pop();
            pq.push(x*2+y);
            res++;
        }
        return pq.top()<k ? -1 : res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {2,11,10,1,3};
    int k = 10;
    cout << s.minOperations(nums, k) << endl;
}
